zcash_client_backend/src/data_api.rs

```diff
@@ -913,6 +920,78 @@ pub trait WalletRead {
     ) -> Result<HashMap<TransparentAddress, NonNegativeAmount>, Self::Error> {
         Ok(HashMap::new())
     }
+
+    /// Returns the metadata associated with a given transparent receiver in an account
+    /// controlled by this wallet.
+    ///
+    /// This is equivalent to (but may be implemented more efficiently than):
+    /// ```no_run
+    /// if let Some(result) = self.get_transparent_receivers(account)?.get(address) {
+    ///     return Ok(result.clone());
```

Comment:
Notice that this can be `Ok(None)` if we recognize the address as a non-ephemeral transparent receiver and don't have its metadata, but that's correct because in that case it can't also be an ephemeral transparent address.